test/functional/services/common/find.ts

```diff
@@ -151,15 +151,23 @@ export async function FindProvider({ getService }: FtrProviderContext) {
       });
     }
 
-    public async allByLinkText(
-      selector: string,
+    public async allByButtonText(
+      buttonText: string,
+      element: WebDriver | WebElement | WebElementWrapper = driver,
       timeout: number = defaultFindTimeout
-    ): Promise<WebElementWrapper[]> {
-      log.debug(`Find.allByLinkText('${selector}') with timeout=${timeout}`);
-      await this._withTimeout(timeout);
-      const elements = await driver.findElements(By.linkText(selector));
-      await this._withTimeout(defaultFindTimeout);
-      return wrapAll(elements);
+    ): Promise<string[]> {
+      log.debug(`Find.byButtonText('${buttonText}') with timeout=${timeout}`);
+      return await retry.tryForTime(timeout, async () => {
+        // tslint:disable-next-line:variable-name
+        const _element = element instanceof WebElementWrapper ? element._webElement : element;
+        const allButtons = wrapAll(await _element.findElements(By.tagName('button')));
+        const buttonTexts = await Promise.all(
+          allButtons.map(async (el) => {
+            return el.getVisibleText();
+          })
+        );
+        return buttonTexts.filter((text) => text.trim() === buttonText.trim());
+      });
```

Comment:
I think it will be better to keep the original implementation as-is: `allByLinkText` is searching using `By.linkText` locator strategy. Just in case smb will use it in the future.

Having `allByButtonText` sounds good, but there is one thing that may bring flakiness:
`findElements` is also using `defaultFindTimeout` as timeout for searching elements: it means up to 10 sec to find smth.
So in case `find.allByButtonText(locator)` is called and there is no buttons at all, the retry will fail on timeout before reaching line 164.

I don't have a nice idea at the moment, but it probably makes sense to do `await this._withTimeout(0);` before retry, still important to set it back `await this._withTimeout(defaultFindTimeout);` on the finish (even if retry fails)

Reply:
Sorry, it probably was a little presumptuous deleting that function. I've brought it back and made the changes suggested!